src/application.ts

```diff
@@ -37,34 +55,34 @@ export class Application extends NamedDefaultableInMemoryEntity {
     }
 
     getBuilds() {
-        const data = getAppData(this.prop("name"));
+        const data = getAppData(this.prop("name")) as ApplicationData;
         const { versions } = data;
         const builds = ["Default"];
         versions.map((v) => v.build && builds.push(v.build));
         return lodash.uniq(builds);
     }
 
     getVersions() {
-        const data = getAppData(this.prop("name"));
+        const data = getAppData(this.prop("name")) as ApplicationData;
         const { versions } = data;
-        const these = versions.map((v) => v.version);
+        const these: string[] = versions.map((v) => v.version);
         return lodash.uniq(these);
     }
 
     static getUniqueAvailableNames() {
         return allApplications;
     }
 
-    getExecutableByName(name = null) {
-        return new this.constructor.Executable(
+    getExecutableByName(name?: string) {
+        return new Application.Executable(
```

Comment:
Did you check that without `this.constructor` re-defining Executable in webapp startup will still work correctly??

Reply:
Seems to work as expected